hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpMessageDispatcherService.java

```diff
@@ -284,12 +284,19 @@ private static EventTopic getEventTypeForAction(final Action action) {
      */
     @EventListener(classes = CancelTargetAssignmentEvent.class)
     protected void targetCancelAssignmentToDistributionSet(final CancelTargetAssignmentEvent cancelEvent) {
-        if (isNotFromSelf(cancelEvent)) {
+        if (!shouldBeProcessed(cancelEvent)) {
             return;
         }
 
-        sendCancelMessageToTarget(cancelEvent.getTenant(), cancelEvent.getEntity().getControllerId(),
-                cancelEvent.getActionId(), cancelEvent.getEntity().getAddress());
+        final Target target = cancelEvent.getEntity();
+        if (target != null) {
+            sendCancelMessageToTarget(cancelEvent.getTenant(), target.getControllerId(), cancelEvent.getActionId(),
+                    target.getAddress());
+        } else {
+            LOG.debug(
```

Comment:
Please log at least on info level because the event isn't processed. Not sure if it should be a warning.